core/src/main/resources/jenkins/model/CoreEnvironmentContributor/buildEnv.properties

```diff
@@ -3,6 +3,7 @@ BUILD_ID.blurb=The current build ID, identical to BUILD_NUMBER for builds create
 BUILD_DISPLAY_NAME.blurb=The display name of the current build, which is something like "#153" by default.
 JOB_NAME.blurb=Name of the project of this build, such as "foo" or "foo/bar".
 JOB_BASE_NAME.blurb=Short Name of the project of this build stripping off folder paths, such as "foo" for "bar/foo".
+JOB_FOLDER_NAME.blurb=Parent directory path of the project of this build, such as "bar" for "bar/foo".
```

Comment:
Terminology inconsistent after rename. Perhaps "Full name of the parent folder of the current job"?

Reply:
Good point; I've changed it to 'Full name of the parent folder of this build, such as "bar" for "bar/foo".' to be consistent with the other messages.

Reply:
We distinguish between projects and builds, and the new version of this message does not. See the message for JOB_BASE_NAME: "Name of the project of this build".

(Builds can be hierarchical (Maven and Matrix job types do this), but it's a parallel hierarchy to corresponding jobs.)